controllers/progressivesync_controller.go

```diff
@@ -267,6 +280,67 @@ func (r *ProgressiveSyncReconciler) getClustersFromSelector(ctx context.Context,
 	return secrets, nil
 }
 
+// Calculate the hash value from the the source ref object (Spec of the ApplicationSet)
+// Store hash value inside the progressive sync annotation map
+
+func (r *ProgressiveSyncReconciler) updatePsHashAnnotation(ctx context.Context, pr *syncv1alpha1.ProgressiveSync, newHashValue string) error {
+	retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+
+		key := client.ObjectKeyFromObject(pr)
+		latest := syncv1alpha1.ProgressiveSync{}
+		if err := r.Client.Get(ctx, key, &latest); err != nil {
+			return err
+		}
+
+		if len(latest.Annotations) == 0 {
+			latest.Annotations = make(map[string]string)
+		}
+
+		val := latest.Annotations["specHashValue"]
+		if val != newHashValue {
+			r.Log.Info("Setting the annotation for the new hash value")
+			latest.Annotations["specHashValue"] = newHashValue
+			if err := r.Client.Update(ctx, &latest); err != nil {
+				return err
+			}
+		}
+		return nil
+	})
+	return retryErr
+}
+
+func (r *ProgressiveSyncReconciler) calculateHashValue(ctx context.Context, pr *syncv1alpha1.ProgressiveSync) (string, error) {
+	key := client.ObjectKeyFromObject(pr)
+	latest := syncv1alpha1.ProgressiveSync{}
+	if err := r.Client.Get(ctx, key, &latest); err != nil {
+		return "", err
+	}
+
+	////////////////////////////////////////////////
+	// Using a unstructured object.
+	//u := unstructured.Unstructured{}
+	//_ = r.Client.Get(ctx, client.ObjectKey{
+	//	Namespace: "argocd",
+	//	Name:      latest.Spec.SourceRef.Name,
+	//}, &u)
+	//
+	//hashValue := fnv.New32a()
+	//hash.DeepHashObject(hashValue, u)
+	//
+	//return rand.SafeEncodeString(fmt.Sprint(hashValue.Sum32())), nil
+	////////////////////////////////////////////////
+
+	appSet := applicationset.ApplicationSet{}
+	if err := r.Client.Get(ctx, client.ObjectKey{Namespace: "argocd", Name: latest.Spec.SourceRef.Name}, &appSet); err != nil {
+		r.Log.Info("Failed to retrieve the ApplicationSet object")
+	}
+
+	hashValue := fnv.New32a()
+	hash.DeepHashObject(hashValue, appSet.Spec)
```

Comment:
Don't we need to hash on _both_ the appset and the progressivesync?
If we change the progressivesync parameters it will be considered as a new deployment? (even though, granted, all the apps should be already sync'ed)

Reply:
The ProgressiveSync CRD tells you _how_ to sync the applications, so I think we should only hash the ApplicationSet which is the _what_.

Reply:
We had a chat about this and we agreed to look into adding the ProgressiveSync as well, once we have fixed the issue around how we do the calculations with percentages. And also we agreed that it can be done as a separate PR once the fix is in place. Seb, I will resolve for now, but please let us know if there is something you want to discuss.